### PR TITLE
[1.x] Use `__invoke` for single action controllers and group common middleware

### PR DESCRIPTION
**Pull request by dammy001**

### CODE REVIEW

**Comment by timacdonald** [comment]
If we do this we probably want to rename the controllers to show what the action is that the take.

e.g. `NewPasswordController` > `StoreNewPasswordController` etc.

I'd prefer we only utilised single action controllers for controllers that land outside of the REST actions and fallback to a more RPC style approach. Just my thoughts.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /vendor
+.idea
```
Comment (jessarcher) [inline .gitignore:2]: This should really be part of your own global `gitignore` as it's not a file generated by this repo.

https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/

Comment (dammy001) [reply .gitignore]: Thanks

**Comment by taylorotwell** [comment]
I'm fine with how it is for now.

**Pull request closed without merging**